lib/web3/agentkit/wallet-providers/privyWalletProvider.ts

```diff
@@ -58,4 +82,161 @@ export class PrivyWalletProvider extends ViemWalletProvider {
   getName(): string {
     return "privy_wallet_provider";
   }
+
+  /**
+   * Signs a message.
+   *
+   * @param message - The message to sign.
+   * @returns The signed message.
+   */
+  async signMessage(message: string | { raw: `0x${string}` }): Promise<`0x${string}`> {
```

Comment:
If you can't tell. Instead of extending ViemWalletProvider (from agentkit) I just added all the methods here then changed this one method so that you can pass the { raw: `0x${string}` } object in. This was required to get a valid signature. It seems ludicrous that CB doesn't support it out of the box since Viem does.